packages/apps/shopify-api/lib/session/session.ts

```diff
@@ -134,6 +146,48 @@ export class Session {
     return session;
   }
 
+  public static async fromEncryptedPropertyArray(
+    entries: SessionParamsArray,
+    cryptoKey: CryptoKey,
+    returnUserData = false,
+  ) {
+    const decryptedEntries: SessionParamsArray = [];
+    for (const [key, value] of entries) {
+      switch (key) {
+        case 'accessToken':
+          decryptedEntries.push([
+            key,
+            await this.decryptValue(value as string, cryptoKey),
+          ]);
+          break;
+        default:
+          decryptedEntries.push([key, value]);
+          break;
+      }
+    }
+
+    return this.fromPropertyArray(decryptedEntries, returnUserData);
+  }
+
+  private static async encryptValue(value: string, key: CryptoKey) {
+    const iv = generateIV();
+    const cipher = await encryptString(value, {key, iv});
+
+    return `${Session.CIPHER_PREFIX}${asBase64(iv)}${cipher}`;
+  }
+
+  private static async decryptValue(value: string, key: CryptoKey) {
+    if (!value.startsWith(Session.CIPHER_PREFIX)) {
+      return value;
+    }
+
+    const keyString = value.slice(Session.CIPHER_PREFIX.length);
```

Comment:
Nit: Would it make sense to put getting the IV and cipher into the encrypt helpers files? Like how we have generateIV()

Reply:
Yeah I guess that would make sense - I don't see us encrypting anything other than sessions but it would keep the session clean. Let's do it.